From 18d5010c9ee3817ac0475c1d634c2115ba7485db Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Julian=20R=C3=BCth?= <julian.rueth@fsfe.org>
Date: Tue, 23 Aug 2022 17:21:37 +0530
Subject: [PATCH] Simplify cmake setup

we only want to build the tool and nothing else but these checks are
trying to build other bits somehow anyway.
---
 CMakeLists.txt | 42 +-----------------------------------------
 1 file changed, 1 insertion(+), 41 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1dceac8..a7e16b3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -5,30 +5,6 @@
 cmake_minimum_required(VERSION 3.11)
 project(cppast VERSION 0.1.0)
 
-# options
-option(CPPAST_BUILD_TEST "whether or not to build the tests" OFF)
-option(CPPAST_BUILD_EXAMPLE "whether or not to build the examples" OFF)
-option(CPPAST_BUILD_TOOL "whether or not to build the tool" OFF)
-
-if(${CPPAST_BUILD_TEST} OR (CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR))
-    set(build_test ON)
-    set(CMAKE_EXPORT_COMPILE_COMMANDS ON) # for the self integration test
-else()
-    set(build_test OFF)
-endif()
-
-if(${CPPAST_BUILD_EXAMPLE} OR (CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR))
-    set(build_example ON)
-else()
-    set(build_example OFF)
-endif()
-
-if(${CPPAST_BUILD_TOOL} OR (CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR))
-    set(build_tool ON)
-else()
-    set(build_tool OFF)
-endif()
-
 find_package(type_safe REQUIRED)
 find_package(Threads REQUIRED)
 
@@ -43,21 +19,5 @@ add_definitions(${LLVM_DEFINITIONS_LIST})
 
 llvm_map_components_to_libnames(llvm_libs support core irreader)
 
-if(build_test AND CPPAST_TEST_GCOV AND (CMAKE_CXX_COMPILER_ID STREQUAL "GNU"))
-    list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake")
-    include(CodeCoverage)
-    APPEND_COVERAGE_COMPILER_FLAGS()
-endif()
-
 add_subdirectory(src)
-
-if(${build_test})
-    enable_testing()
-    add_subdirectory(test)
-endif()
-if(${build_example})
-    add_subdirectory(example)
-endif()
-if(${build_tool})
-    add_subdirectory(tool)
-endif()
+add_subdirectory(tool)
-- 
2.37.1

